packages/metro-config/src/defaults/exclusionList.js

```diff
@@ -15,10 +15,19 @@ var list = [/website\/node_modules\/.*/, /.*\/__tests__\/.*/];
 
 function escapeRegExp(pattern) {
   if (Object.prototype.toString.call(pattern) === '[object RegExp]') {
-    return pattern.source.replace(/\//g, path.sep);
+    // the forward slash may or may not be escaped in regular expression depends
+    // on if it's in brackets. eg. /foo\/bar/ and /[/\\]foo/ are both valid.
```

Comment:
This example is somewhat confusing because there are multiple things that changes (bar no longer present, using `\\`). I think what you meant is that

`/foo\/bar/` and `/foo[/]bar/` are equal

Reply:
My comments weren't clear enough. Regular expression is confusing!

What I want to say here is the forward slash is what causing the trouble for our original code. If it's in brackets like `/[/\\]foo/`, in node v10, it will automatically be converted to `/[\/\\]foo/`. However,  in node v12, it will not be coverted at all (you can try that in https://www.jdoodle.com/ia/3tF).

If the forward slash is outside of brackets, as it will mess up with the regular expression start/end character, it will always need to be escaped. Hence, node v10 won't escape it again. That is to say, `/foo\/bar/` will not be converted in both node versions.

Hence, our code should properly handle both cases by the `|` operator. Since regular expression is greedy in matching patterns, it will automatically cover both cases. Assuming `const a = /foo\/bar/;` and `const b = /[/\\]foo/;`.

node 12:
```
a.source foo\/bar
a in windows foo\\bar
a in linux foo\/bar
b.source [/\\]foo
b in windows [\\\\]foo
b in linux [\/\\]foo
```

node 10
```
a.source foo\/bar
a in windows foo\\bar
a in linux foo\/bar
b.source [\/\\]foo
b in windows [\\\\]foo
b in linux [\/\\]foo
```